pkg/handler/topology.go

```diff
@@ -48,68 +58,106 @@ func GetTopology(cfg *config.Config) func(w http.ResponseWriter, r *http.Request
 	}
 }
 
-func getTopologyFlows(cfg *config.Config, client httpclient.Caller, params url.Values) (*model.AggregatedQueryResponse, int, error) {
-	hlog.Debugf("GetTopology query params: %s", params)
+func (h *Handlers) extractTopologyQueryParams(params url.Values) (*loki.TopologyInput, filters.MultiQueries, v1.Range, int, error) {
+	in := loki.TopologyInput{DedupMark: h.Cfg.Frontend.Deduper.Mark}
+	qr := v1.Range{}
+	var reqLimit int
+	var err error
 
-	start, err := getStartTime(params)
+	in.Start, qr.Start, err = getStartTime(params)
 	if err != nil {
-		return nil, http.StatusBadRequest, err
+		return nil, nil, qr, reqLimit, err
 	}
-	end, err := getEndTime(params)
+	in.End, qr.End, err = getEndTime(params)
 	if err != nil {
-		return nil, http.StatusBadRequest, err
+		return nil, nil, qr, reqLimit, err
 	}
-	limit, reqLimit, err := getLimit(params)
+	in.Top, reqLimit, err = getLimit(params)
 	if err != nil {
-		return nil, http.StatusBadRequest, err
+		return nil, nil, qr, reqLimit, err
 	}
-	rateInterval, err := getRateInterval(params)
+	in.RateInterval, err = getRateInterval(params)
 	if err != nil {
-		return nil, http.StatusBadRequest, err
+		return nil, nil, qr, reqLimit, err
 	}
-	step, err := getStep(params)
+	in.Step, qr.Step, err = getStep(params)
```

Comment:
why introduced `qr.Step` I don't see it used ?

Reply:
It's part of the prometheus model, used in the prom library, not directly from our code. The whole QueryRange object is passed in a lib call here: https://github.com/netobserv/network-observability-console-plugin/pull/516/files#diff-9994e7f5220cdeef1b3834c4ee83d0225747947c8abf3fddb2c6a68165805e4fR53